spec/requests/client_visibility_spec.rb

```diff
@@ -5,839 +5,896 @@
+        before do
+          user.roles << can_search_window
+          sign_in user
+        end
+        it 'user can see only window clients' do
+          get clients_path(q: 'bob')
+          doc = Nokogiri::HTML(response.body)
+          expect(doc.text).to include('Displaying 1 client')
+          expect(response).to have_http_status(200)
+        end
+        it 'user cannot see client dashboard' do
+          get client_path(window_destination_client)
+          expect(response).to redirect_to(user.my_root_path)
+        end
+        it 'user can see directory information for window client' do
+          get simple_client_path(window_source_client)
+          expect(response).to have_http_status(200)
+        end
       end
-      describe 'and the user is assigned a data source' do
+      describe 'and the user has a role granting visibility by data source' do
         before do
-          user.add_viewable(non_window_visible_data_source)
+          user.roles << can_see_clients_in_window_for_assigned_data_sources
+          user.roles << can_search_window
+          sign_in user
         end
-        it 'user can see one client in expected data source and any window clients' do
+        it 'user can see window clients in search results' do
           get clients_path(q: 'bob')
           doc = Nokogiri::HTML(response.body)
-          expect(doc.text).to include('Displaying all 2 clients')
+          expect(doc.text).to include('Displaying 1 client')
           expect(response).to have_http_status(200)
         end
+        describe 'and the user is assigned a data source' do
+          before do
+            user.add_viewable(non_window_visible_data_source)
+          end
+          it 'user can see one client in expected data source and any window clients' do
+            get clients_path(q: 'bob')
+            doc = Nokogiri::HTML(response.body)
+            expect(doc.text).to include('Displaying all 2 clients')
+            expect(response).to have_http_status(200)
+          end
+        end
       end
     end
-  end
-
-  context 'when config s is in affect' do
-    before do
-      GrdaWarehouse::Config.delete_all
-      GrdaWarehouse::Config.invalidate_cache
-    end
-    let!(:config) { create :config_s }
-    let!(:user) { create :user }
 
-    describe 'and the user does not have a role' do
-      it 'user cannot see any clients' do
-        get clients_path(q: 'bob')
-        expect(response).to redirect_to(new_user_session_path)
-      end
-    end
-    describe 'and the user has a role granting can view clients' do
+    context 'when config s is in affect' do
       before do
-        user.roles << can_view_clients
-        user.roles << can_search_window
-        sign_in user
+        GrdaWarehouse::Config.delete_all
+        GrdaWarehouse::Config.invalidate_cache
+      end
+      # rubocop:disable Layout/ArgumentAlignment
+      let!(:config) do
+        create :config_b,
```

Comment:
As noted above, this should be `:config_s` (not b).
Then the next section should have the tests from the `:config_s` section in the original file.